pandas/_libs/src/ujson/python/objToJSON.c

```diff
@@ -2126,6 +2126,21 @@ double Object_getDoubleValue(JSOBJ Py_UNUSED(obj), JSONTypeContext *tc) {
     return GET_TC(tc)->doubleValue;
 }
 
+const char *Object_getBigNumStringValue(JSOBJ obj, JSONTypeContext *tc, 
+                                    size_t *_outLen) {
+    PyObject* repr = PyObject_Str(obj);
+    PyObject* str = PyUnicode_AsUTF8AndSize(repr, _outLen);
+    char bytes[_outlen];
```

Comment:
This array is managed on the stack, so it automatically gets cleaned up when it goes out of scope, which isn’t what you want (and the reason for multiple memcpy calls)

Instead just malloc _outLen bytes and then memcpy to that, before assign that to cStr

Reply:
> Instead just malloc _outLen bytes and then memcpy to that, before assign that to cStr

Ok, great! I fixed that.

One thing I'm stuck on is converting from the argument of `PyUnicode_AsUTF8AndSize` (which has to be a `Py_ssize_t`) to `_outLen` (which is C `size_t`)...

Also, just to check: I do need dereference `_outLen` in the calls to `memcpy` and `malloc`?